### PR TITLE
Added txtConflict logic

### PR DESCRIPTION
**Pull request by arnoldblinn**
Added txtConflict attributes to TXT records to clarify how conflicts on TXT records should be handled by the DNS Provider.

Also cleaned up and clarified several issues.

### CODE REVIEW

File: Domain Connect Spec Draft.adoc [hunk 1]
```diff
@@ -1550,34 +1585,38 @@ consent UX should warn the user about these conflicts. For templates,
...
 records that would be overwritten. This could be progressively disclosed.
 
-Note: When applying the same template, DNS Providers should not detect
+Note: When re-applying the same template, ideally DNS Providers would not detect
 the conflict. Instead, the first template would be removed and the new
 instance applied. For most templates this is a benign operation, unless
 the template contains variables in host names. For consideration
 of this, see the section below.
 
 === Record Types and Conflicts
```
Comment (pawel-kow) [inline Domain Connect Spec Draft.adoc]: Reading through the changes I now notices, that the whole section is talking about deleting or adding records, while it should be rather referring to detection of situations with or without conflict.

That the conflict resolution leads to deletion of the conflicting records upon new template application is something we describe in the previous chapter, also in a bit vague way. I'll issue a proposal to rephrase the whole thing...

File: Domain Connect Spec Draft.adoc [hunk 2]
```diff
@@ -1550,34 +1585,38 @@ consent UX should warn the user about these conflicts. For templates,
...
 * Replace records of the same type for A, AAAA, MX, CNAME, APEXCNAME,
 SRV. If the template specifies an A or AAAA, the respective A or AAAA
 record should be removed to avoid IPv4 and IPv6 pointing to different
 services.
-* Append to the existing records of the same type for TXT.
-** An exception exists for records of unique nature like SPF or DMARC
--which should be replaced. To avoid conflicts for SPF, it is advisable to use SPFM record type instead (see: <<spf-record-merging>>).
-* Replace any record for CNAME.
+* For TXT records deletion is handled looking at txtConflictMatchingMode
+** None: This indicates that the TXT records should be appended to the zone without removing existing values of the same host.
```
Comment (pawel-kow) [inline Domain Connect Spec Draft.adoc]:
```suggestion
** None: This indicates that the TXT records should be appended to the zone without removing existing values. This is the default behavior.```

File: Domain Connect Spec Draft.adoc [hunk 3]
```diff
@@ -1550,34 +1585,38 @@ consent UX should warn the user about these conflicts. For templates,
...
+** All: This indicates that the TXT records with the same host name are deleted prior to applying.
```
Comment (pawel-kow) [inline Domain Connect Spec Draft.adoc]:
```suggestion
** All: This indicates that all the TXT records are deleted prior to applying.
```

**Comment by pawel-kow** [comment]
Re-write of the chapter issued under https://github.com/Domain-Connect/spec/pull/25

File: Domain Connect Spec Draft.adoc (hunk 3 above)
Comment (pawel-kow) [reply Domain Connect Spec Draft.adoc]: Not actual anymore

File: Domain Connect Spec Draft.adoc (hunk 2 above)
Comment (pawel-kow) [reply Domain Connect Spec Draft.adoc]: Not actual anymore

File: Domain Connect Spec Draft.adoc (hunk 1 above)
Comment (pawel-kow) [reply Domain Connect Spec Draft.adoc]: Not actual anymore

File: Domain Connect Spec Draft.adoc [hunk 4]
```diff
@@ -1422,107 +1438,123 @@ SPFM: host, spfRules
...
 |*Data*
 |String
 |data
-|This is the data for a TXT record in DNS
+|The data for a TXT record in DNS
+
+|*TXT Conflict Matching Mode*
+|String
+|txtConflictMatchingMode
+|Describes how conflicts on the TXT record should be handled. Possible values are
```
[the remark this answers is not in the document]
Comment (arnoldblinn) [reply Domain Connect Spec Draft.adoc]: Done

File: Domain Connect Spec Draft.adoc [hunk 5]
```diff
@@ -1422,107 +1438,123 @@ SPFM: host, spfRules
...
+|*TXT Conflict Matching Mode*
+|String
+|txtConflictMatchingMode
+|Describes how conflicts on the TXT record should be handled. Possible values are 
+None, All, or Prefix. The default value is None. See below.
+
+|*TXT Conflict Matching Prefix*
+|String
+|txtConflictMatchingPrefix
+|The prefix for handling deletes when txtConflictMatchingMode is "Prefix". This
```
[the remark this answers is not in the document]
Comment (arnoldblinn) [reply Domain Connect Spec Draft.adoc]: Done

File: Domain Connect Spec Draft.adoc (hunk 1 above)
Comment (arnoldblinn) [reply Domain Connect Spec Draft.adoc]: I think you re-wrote this.

File: Domain Connect Spec Draft.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (pawel-kow) [reply Domain Connect Spec Draft.adoc]: Done

File: Domain Connect Spec Draft.adoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (pawel-kow) [reply Domain Connect Spec Draft.adoc]: Done

**Pull request merged**